File: app/src/main/java/com/ucb/midterm/news/MainActivity.java
package com.ucb.midterm.news;

import android.content.res.Configuration;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements HeadlineListFragment.OnHeadlineSelectedListener {

    private boolean isLandscape;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        isLandscape = getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE;

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(isLandscape ? R.id.headline_container : R.id.fragment_container, new HeadlineListFragment())
                    .commit();
        }
    }

    @Override
    public void onHeadlineSelected(News selectedNews) {
        NewsContentFragment contentFragment = NewsContentFragment.newInstance(selectedNews.getHeadline(), selectedNews.getContent());

        if (isLandscape) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.content_container, contentFragment)
                    .commit();
        } else {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.fragment_container, contentFragment)
                    .addToBackStack(null)
                    .commit();
        }
    }
}

File: app/src/main/java/com/ucb/midterm/todolist/MainActivity.kt
package com.ucb.midterm.todolist

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var todoList: ArrayList<String>? = null
    private var adapter: TodoAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        todoList = ArrayList()
        val todoListView = findViewById<ListView>(R.id.todoListView)
        val todoInput = findViewById<EditText>(R.id.todoInput)
        val addButton = findViewById<Button>(R.id.addButton)

        adapter = TodoAdapter(this, todoList!!)
        todoListView.adapter = adapter

        addButton.setOnClickListener { view: View? ->
            val todo = todoInput.text.toString()
            if (!todo.isEmpty()) {
                todoList!!.add(todo)
                adapter!!.notifyDataSetChanged()
                todoInput.setText("")
            }
        }
    }
}